includes/functions/functions_traffic.php

```diff
@@ -29,11 +29,11 @@ function zen_get_ip_address() {
             $ip = $_SERVER['HTTP_FORWARDED_FOR'];
         } elseif (isset($_SERVER['HTTP_FORWARDED'])) {
             $ip = $_SERVER['HTTP_FORWARDED'];
-        } else {
+        } elseif (isset($_SERVER['REMOTE_ADDR'])) {
             $ip = $_SERVER['REMOTE_ADDR'];
         }
     }
-    if (trim($ip) == '') {
+    if (empty($ip)) { 
```

Comment:
This processes differently than before. If for some reason `$ip` contained spaces around a value of zero, the previous method would trigger a `truthy` resolve. Using empty on a string of value 0 surrounded by spaces will result in a false response here. If that is intentional, then would suggest providing explanation that includes more than that this is some sort of edge case situation. 

Based on the changes above it seems like this line could remain as it was since it would either be an empty string or whatever set value is to be used.

Reply:
I agree, @mc12345678, how about simply casting the `$ip` value submitted to trim:
```php
if (trim((string)$ip) == '') {
```

Reply:
Good catch.